src/WebJobs.Script/ExtensionBundle/ExtensionBundleManager.cs

```diff
@@ -293,6 +293,38 @@ internal static string FindBestVersionMatch(VersionRange versionRange, IEnumerab
             return matchingVersion?.ToString();
         }
 
+        private NuGetVersion ResolvePlatformReleaseChannelVersion(IList<NuGetVersion> orderedByDescBundles) => _platformReleaseChannel switch
+        {
+            ScriptConstants.StandardPlatformChannelNameUpper or ScriptConstants.ExtendedPlatformChannelNameUpper => GetStandardOrExtendedBundleVersion(orderedByDescBundles),
+            ScriptConstants.LatestPlatformChannelNameUpper or "" => GetLatestBundleVersion(orderedByDescBundles),
+            _ => HandleUnknownPlatformReleaseChannelName(orderedByDescBundles)
+        };
+
+        private NuGetVersion GetStandardOrExtendedBundleVersion(IList<NuGetVersion> orderedByDescBundlesList)
+        {
+            if (orderedByDescBundlesList.Count > 1)
+            {
+                return orderedByDescBundlesList[1]; // These channels should resolve to the version prior to latest. This list is in descending order, which makes latest [0], and prior-to-latest [1].
+            }
+
+            // keep the latest version, log a notice
+            var latest = orderedByDescBundlesList.FirstOrDefault();
+            _logger.LogInformation("Unable to apply platform release channel configuration {platformReleaseChannelName}. Only one matching bundle version is available. {latestBundleVersion} will be used", _platformReleaseChannel, latest);
+            return latest;
+        }
+
+        private static NuGetVersion GetLatestBundleVersion(IList<NuGetVersion> orderedByDescBundlesList)
```

Comment:
Why does this need to be static?

Reply:
Suggested by @kshyju - it doesn't use any instance fields or props and provides some perf benefits.

Reply:
How much perf benefits are we talking here 😭

Reply:
https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1822
The gain might be minor in this specific case. But It's a good idea to stick with that pattern in our code base when possible.

Keep in mind that performance is the result of thousands of small optimizations.